### PR TITLE
CDRIVER-4677 conditionally restore AWS instance profile

### PR DESCRIPTION
**Pull request by kevinAlbs**
# Summary

- Conditionally restore the AWS instance profile after task

Changes verified with [this patch build](https://spruce.mongodb.com/version/650b42742a60ed69ab93574e). 

# Background & Motivation

The description of DRIVERS-2639 describes the motivation.

Changes were copied and modified from https://github.com/mongodb/mongo-python-driver/pull/1218/files

### CODE REVIEW

File: .evergreen/config_generator/components/funcs/restore_instance_profile.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            if [[ ! -f aws_e2e_setup.json ]]; then
+                echo "aws_e2e_setup.json not present ... skipping"
+                exit 0
+            fi
+
+            . ./activate-authawsvenv.sh
+            
+            echo "restoring instance profile ... "
+            # Capture and hide logs on success. Logs may included expected `HTTP Error 404: Not Found` messages when checking for instance profile.
+            if ! ( python ./lib/aws_assign_instance_profile.py 2>&1 >| output.txt ); then
```
Comment (eramongodb) [inline .evergreen/config_generator/components/funcs/restore_instance_profile.py:27]:
```suggestion
            if ! { python ./lib/aws_assign_instance_profile.py 2>&1 >|output.txt }; then
```
Avoid unnecessarily starting a subshell. Disregard if this is deliberate to scope env vars set by aws_assign_instance_profile.py.

File: .evergreen/config_generator/components/funcs/restore_instance_profile.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                echo "reassigning instance profile ... failed"
```
Comment (eramongodb) [inline .evergreen/config_generator/components/funcs/restore_instance_profile.py:28]:
```suggestion
                echo "restoring instance profile ... failed"
```
Consistent terminology.

**Review by eramongodb — APPROVED** [review]
Minor suggestions; otherwise, LGTM.

File: .evergreen/config_generator/components/funcs/restore_instance_profile.py (hunk 1 above)
Comment (kevinAlbs) [reply .evergreen/config_generator/components/funcs/restore_instance_profile.py]: Not deliberate. Applying suggestion.

File: .evergreen/config_generator/components/funcs/restore_instance_profile.py (hunk 2 above)
Comment (kevinAlbs) [reply .evergreen/config_generator/components/funcs/restore_instance_profile.py]: Oops. Thank you for the catch.

**Comment by kevinAlbs** [comment]
Latest changes tested with [patch build](https://spruce.mongodb.com/version/650c38023627e0634ec465be).

**Pull request merged**